packages/ra-core/src/dataProvider/withLifecycleCallbacks.ts

```diff
@@ -0,0 +1,590 @@
+                );
+            }
+
+            return result;
+        },
+
+        delete: async function <RecordType extends RaRecord = any>(
+            resource: string,
+            params: DeleteParams<RecordType>
+        ) {
+            let newParams = params;
+            const beforeDeleteHandlers = handlers.filter(
+                h => h.resource === resource && h.beforeDelete
+            );
+            for (let handler of beforeDeleteHandlers) {
+                newParams = await handler.beforeDelete(newParams, dataProvider);
+            }
+
+            let result = await dataProvider.delete<RecordType>(
+                resource,
+                newParams
+            );
+
+            const afterDeleteHandlers = handlers.filter(
+                h => h.resource === resource && h.afterDelete
+            );
+            for (let handler of afterDeleteHandlers) {
+                result = await handler.afterDelete(result, dataProvider);
+            }
+
+            return result;
+        },
+
+        updateMany: async function <RecordType extends RaRecord = any>(
+            resource: string,
+            params: UpdateManyParams<RecordType>
+        ) {
+            let newParams = params;
+            const beforeUpdateManyHandlers = handlers.filter(
+                h => h.resource === resource && h.beforeUpdateMany
+            );
+            for (let handler of beforeUpdateManyHandlers) {
+                newParams = await handler.beforeUpdateMany(
+                    newParams,
+                    dataProvider
+                );
+            }
+            const beforeSaveHandlers = handlers.filter(
+                h => h.resource === resource && h.beforeSave
+            );
+            for (let handler of beforeSaveHandlers) {
+                newParams.data = await handler.beforeSave(
+                    newParams.data,
+                    dataProvider
+                );
+            }
+
+            let result = await dataProvider.updateMany<RecordType>(
+                resource,
+                newParams
+            );
+
+            const afterUpdateManyHandlers = handlers.filter(
+                h => h.resource === resource && h.afterUpdateMany
+            );
+            for (let handler of afterUpdateManyHandlers) {
+                result = await handler.afterUpdateMany(result, dataProvider);
+            }
+            const afterSaveHandlers = handlers.filter(
+                h => h.resource === resource && h.afterSave
+            );
+            if (afterSaveHandlers.length > 0) {
+                const { data: records } = await dataProvider.getMany(resource, {
+                    ids: result.data,
+                });
+
+                for (let handler of afterSaveHandlers) {
+                    await Promise.all(
+                        records.map(record =>
+                            handler.afterSave(record, dataProvider)
+                        )
+                    );
```

Comment:
Probably missing an affectation here

Reply:
Nope, `updateMany` doesn't return the updated records, just their ids.